amstrax/auto_processing/copy_live.py

```diff
@@ -0,0 +1,83 @@
+import subprocess
+import pymongo
+import amstrax
+import argparse
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description='Script that automatically copies new runs to stoomboot')
+    parser.add_argument(
+        '--detector',
+        type=str,
+        help='The detector that you are using',
+        default='xams')
+    parser.add_argument(
+        '--max_runs',
+        type=int,
+        help='How many runs you want to copy every time',
+        default=1)
+    args = parser.parse_args()
+    detector = args.detector
+    max_runs = args.max_runs
+
+    runsdb = amstrax.get_mongo_collection()
+    max_runs = 1
+    dest_loc = f'/data/xenon/{detector}/live_data'
+
+    # Take last 'max_runs' runs and sort them
+    runs = list(run_id.get('number', 'no_number')
+                for run_id in
+                runsdb.find({},
+                            projection={'number': 1, '_id': 0}
+                            ).sort('number', pymongo.DESCENDING)[:max_runs])
+
+    rundata = list(run_id.get('data', 'no_data')
+                   for run_id in
+                   runsdb.find({},
+                               projection={'data': 1, '_id': 0}
+                               ).sort('number', pymongo.DESCENDING)[:max_runs])
```

Comment:
*kind of important* mongo queries are so confusing, but this is much better than what I was able to do when I started.

However, you are doing the same brute force query twice to our poor DAQ machine. You can simplify this by doing:

```python
query = {}  # TODO for now, but we should query on the data field in the future
rundocs = list(runsdb.find(query, projection = {'number:1, data:1, _id:1} ))
```

This will simplify this line as follows:
```python
# was
    for (run, location) in zip(runs, live_location):
# will be
    for rd in rundocs:
        run = rd.get('number')
        data_fields = rd.get('data')
        location = None
        for doc in data_fields:
            if doc['type'] == 'live':  
               location = doc['location']
        if run is None or location is None: 
             print(f'For {run} we got no data? Rundoc: {rd}')
```